interfaces/repo.go

```diff
@@ -438,7 +448,7 @@ func (r *Repository) ResolveDisconnect(plugSnapName, plugName, slotSnapName, slo
 
 // Connect establishes a connection between a plug and a slot.
 // The plug and the slot must have the same interface.
-func (r *Repository) Connect(ref ConnRef) error {
+func (r *Repository) Connect(ref ConnRef, plugAttrs map[string]interface{}, slotAttrs map[string]interface{}) error {
```

Comment:
shouldn't this or somewhere call Sanitize*  on the dynamic attrs, that's the place that can set default value afaiu

also we should remember the issue of not passing nil  to methods that can write on attrs